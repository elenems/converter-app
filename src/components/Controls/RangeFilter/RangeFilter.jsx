import React from "react";
import "date-fns";
import DateFnsUtils from "@date-io/date-fns";
import {
  KeyboardDatePicker,
  MuiPickersUtilsProvider
} from "@material-ui/pickers";
export default function RangeFilter(props) {
  const [selectedDate, setSelectedDate] = React.useState(
    new Date(props.historyRange)
  );
  function handleDateChange(date) {
    setSelectedDate(date);
    props.historyChange(date);
  }

  return (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <KeyboardDatePicker
        disableToolbar
        variant="inline"
        format="yyyy/MM/dd"
        margin="normal"
        id="date-picker-inline"
        label="Date picker inline"
        value={selectedDate}
        onChange={handleDateChange}
        KeyboardButtonProps={{
          "aria-label": "change date"
        }}
      />
    </MuiPickersUtilsProvider>
  );
}
