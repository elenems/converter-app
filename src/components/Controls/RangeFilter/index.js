import RangeFilter from './RangeFilter.jsx';
export default RangeFilter;