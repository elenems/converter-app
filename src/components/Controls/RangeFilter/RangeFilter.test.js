import React from 'react';
import { shallow } from 'enzyme';
import RangeFilter from './RangeFilter';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('RangeFilter', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<RangeFilter/>).debug();
    expect(component).toMatchSnapshot();
  });
});