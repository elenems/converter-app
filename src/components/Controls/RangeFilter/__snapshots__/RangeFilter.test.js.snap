// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeFilter should render correctly in "debug" mode 1`] = `
"<MuiPickersUtilsProvider utils={[Function: DateFnsUtils]}>
  <PickerWithState disableToolbar={true} variant=\\"inline\\" format=\\"yyyy/MM/dd\\" margin=\\"normal\\" id=\\"date-picker-inline\\" label={[undefined]} value={{...}} onChange={[Function: handleDateChange]} KeyboardButtonProps={{...}} minDate={{...}} maxDate={{...}} invalidDateMessage=\\"Invalid Date Format\\" minDateMessage=\\"Date should not be before minimal date\\" maxDateMessage=\\"Date should not be after maximal date\\" allowKeyboardControl={true} openTo=\\"date\\" views={{...}} />
</MuiPickersUtilsProvider>"
`;
